### ISSUE TITLE
Error executing action `update` on resource 'php_pear_channel[pear.php.net]'

### ISSUE THREAD

**Issue by gbradner** [chef-cookbooks/php#198]
### Cookbook version
3.1.0

### Chef-client version
Chef: 12.6.0

### Platform Details
Ubuntu 14.04.5 LTS

### Scenario:
chef-client run

### Steps to Reproduce:
run chef-client
pear is in path

### Expected Result:
successful run

### Actual Result:
```
Recipe: php::default
  * php_pear_channel[pear.php.net] action update

    ================================================================================
    Error executing action `update` on resource 'php_pear_channel[pear.php.net]'
    ================================================================================

    NoMethodError
    -------------
    No resource or method named `exists?' for `Custom resource php_pear_channel from cookbook php action provider "pear.php.net"'

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/php/resources/pear_channel.rb:58:in `block in class_from_file'
 /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
 /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
 /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
 /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
 /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/php/recipes/default.rb

     25: php_pear_channel 'pear.php.net' do
     26:   action :update
     27: end
     28:

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/php/recipes/default.rb:25:in `from_file'

    php_pear_channel("pear.php.net") do
      action [:update]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :php_pear_channel
      cookbook_name "php"
      recipe_name "default"
    end

Running handlers:
[2017-04-17T19:47:39+00:00] ERROR: Running exception handlers
Running handlers complete
[2017-04-17T19:47:39+00:00] ERROR: Exception handlers complete
Chef Client failed. 7 resources updated in 41 seconds
[2017-04-17T19:47:39+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2017-04-17T19:47:39+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-04-17T19:47:39+00:00] **ERROR: php_pear_channel[pear.php.net] (php::default line 25) had an error: NoMethodError: No resource or method named `exists?' for `Custom resource php_pear_channel from cookbook php action provider "pear.php.net"'**
[2017-04-17T19:47:39+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by tas50** [comment]
This has been fixed on master and I'll have a release going out today or tomorrow with the fix

**Comment by gsdevme** [comment]
@tas50 Did this get fixed? Im on Chef 12.6.0 + CentOS 6.8

```
  php (4.2.0)
    build-essential (>= 0.0.0)
    mysql (>= 6.0.0)
    xml (>= 0.0.0)
    yum-epel (>= 0.0.0)
```
